frontend/ui/widget/virtualkeyboard.lua

```diff
@@ -310,6 +310,17 @@ function VirtualKey:init()
     self.flash_keyboard = G_reader_settings:nilOrTrue("flash_keyboard")
 end
 
+function VirtualKey:paintTo(...)
+    InputContainer.paintTo(self, ...)
+
+    -- Fudge self.dimen to include the padding, to make sure said padding is covered by our ges_events range...
+    local padding = bit.rshift(self.keyboard.key_padding, 1) -- truncated div by 2
```

Comment:
So, that's a floor() and not a ceil(), so if key_padding=3, we'll get padding=1 and there will be 1px between the tap zones ?

Reply:
Yup, I had no idea how overlaps would behave, so I opted for truncation.

Reply:
> I had no idea how overlaps would behave

Badly ;p. It's FIFO, so, the leftmost (or topmost) key wins if there's an overlap.

Reply:
and what matters is that _any_ of them wins, no matter which, so a tap is not ignored.

Reply:
Hmm, no clue why width & height need a double-dose, though, going to check on the emu...

Reply:
> Hmm, no clue why width & height need a double-dose, though, going to check on the emu...

Uh, because maths, duh'. Must be getting old :D.

Reply:
There, temporary brainfart has passed; now handles odd gaps sanely and without overlaps ;p.